tools/qc/fastq_utils/fastq_pre_barcodes.xml

```diff
@@ -0,0 +1,246 @@
+            <option value="64">64</option>
+        </param>
+        <param name="min_qual" label="Minimum Quality" optional='true' value='' argument="--min_qual" type="integer" min="0" max="40"   help="[0-40]. Defines the minimum quality that all bases in the UMI, Cell or Sample should have (reads that do not pass the criteria are discarded). 0 disables the filter."/>
+        <param label="Interleaved Data" name="interleaved" argument="--interleaved" type="text"   help="Interleaved data, in this format: (read1|read2|index1|index2|index3),(read1|read2|index1|index2|index3)"/>
+        <param label="UMI read"  name="umi_read" argument="--umi_read" type="text"   help="File in which UMI read can be found, in this format: (read1|read2|index1|index2|index3)"/>
+        <param label="UMI offset"  name="umi_offset" argument="--umi_offset" type="integer"   help="Offset (integer)"/>
+        <param label="UMI Size"  name="umi_size" argument="--umi_size" type="integer"   help="Number of bases after the offset"/>
+        <param label="Cell Read"  name="Cell_read" argument="--Cell_read" type="text"   help="File in which Cell can be found, in this format: (read1|read2|index1|index2|index3)"/>
+        <param label="Cell Offset"  name="Cell_offset" argument="--Cell_offset" type="integer"   help="Offset"/>
+        <param label="Cell Size"  name="Cell_size" argument="--Cell_size" type="integer"   help="Number of bases after the offset"/>
+        <param label="Sample Read"  name="sample_read" argument="--sample_read" type="text"   help="File in which sample barcode can be found, in this format: (read1|read2|index1|index2|index3)"/>
+        <param label="Sample Offset"  name="sample_offset" argument="--sample_offset" type="integer"   help="Offset"/>
+        <param label="Sample Size"  name="sample_size" argument="--sample_size" type="integer"   help="Number of bases after the offset"/>
+        <param label="read1 Offset"  name="read1_offset" argument="--read1_offset" type="integer"   help="None"/>
+        <param label="read1 Size"  name="read1_size" argument="--read1_size" type="integer"   help="None"/>
+        <param label="read2 Offset"  name="read2_offset" argument="--read2_offset" type="integer"   help="None"/>
+        <param label="read2 Size"  name="read2_size" argument="--read2_size" type="integer"   help="None"/>
+        <param label="Use 10x tags"  name="use_10x" argument="--10x" type="text"   help="Use 10X UMI tags (UB and UY) instead of the default tags defined in the SAM specification"/>
+        <param label="sam" name="sam" argument="--sam" type="text" help="No available description"/>
+    </inputs>
+    <outputs>
+        <data label="${tool.name} on ${on_string}: Output file 1" name="outfile1" format='?' />
+        <data label="${tool.name} on ${on_string}: Output file 2" name="outfile2" format='?' />
+        <data label="${tool.name} on ${on_string}: Output file 3" name="outfile3" format='?' />
```

Comment:
also format needs to be set here, please see galaxy datatypes in the Galaxy docs.

Reply:
Done in 3896d3d2f77c47f9eca79bff42e7df7e12bbfdef.